src/backend/modules/transaction/transaction.service.ts

```diff
@@ -0,0 +1,282 @@
+                  },
+                  User: {
+                    update: {
+                      balanceCents: {
+                        increment: cost
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }),
+
+          this.prisma.userStock.upsert({
+            where: {
+              userStockId: buyerStockId
+            },
+            update: {
+              stockQuantity: {
+                increment: quantity
+              },
+            },
+            create: {
+              userId: buyOffer.userId,
+              stockQuantity: quantity,
+              companyId: buyOffer.companyId
+            }
+          }),
+
+          this.prisma.transaction.create({
+            data: {
+              buyOfferId: buyOffer.buyOfferId,
+              sellOfferId: sellOffer.sellOfferId,
+              status: 5,
+              sellerId: sellOffer.userId,
+              buyerId: buyOffer.userId,
+            },
+          }),
+        ])
+
+        quantity = 0
+      } else { //Triggers when number of purchased stock is greater then number of sold stock
+        const cost = sellOffer.unitSellPriceCents * sellOffer.quantity
+
+        //Database transaction
+        await this.prisma.$transaction([
+          this.prisma.buyOffer.update({
+            where: {
+              buyOfferId: buyOffer.buyOfferId
+            },
+            data: {
+              status: 0,
+              quantity: {
+                decrement: sellOffer.quantity
+              },
+              User: {
+                update: {
+                  balanceCents: {
+                    decrement: cost
+                  }
+                },
+              }
+            }
+          }),
+
+          this.prisma.sellOffer.update({
+            where: {
+              sellOfferId: sellOffer.sellOfferId
+            },
+            data: {
+              quantity: {
+                decrement: sellOffer.quantity
+              },
+              status: 4,
+              UserStock: {
+                update: {
+                  stockQuantity: {
+                    decrement: sellOffer.quantity
+                  },
+                  User: {
+                    update: {
+                      balanceCents: {
+                        increment: cost
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }),
+
+          this.prisma.userStock.upsert({
+            where: {
+              userStockId: buyerStockId
+            },
+            update: {
+              stockQuantity: {
+                increment: sellOffer.quantity
+              },
+            },
+            create: {
+              userId: buyOffer.userId,
+              stockQuantity: sellOffer.quantity,
+              companyId: buyOffer.companyId
+            }
+          }),
+
+          this.prisma.transaction.create({

```

Comment:
If this call and previous one are identical, then maybe they should be extracted to function?
Even not inside class but "findMatchingSellOffers" function?